sqlmesh/core/engine_adapter/base.py

```diff
@@ -784,16 +823,25 @@ def clone_table(
             )
         )
 
-    def drop_table(self, table_name: TableName, exists: bool = True) -> None:
+    def drop_table(self, table_name: TableName, exists: bool = True, kind: str = "TABLE") -> None:
```

Comment:
I think this is leaking engine implementation details to the outside. Eg. a caller of the `drop_table ` now need to worry about whether it's a "DYNAMIC TABLE" or something else. 

If you want to reuse this API to implement `drop_engine_managed_table`, then I'd suggest having a "private" method that does take the `kind` argument and is called by both `drop_table` and `drop_engine_managed_table`.

Reply:
Yeah this did feel a bit awkward.

I've reinstated the previous public API and created a new private method for dropping table-like objects